app/controllers/pages_controller.rb

```diff
@@ -1,9 +1,17 @@
 # frozen_string_literal: true
 
 class PagesController < SecureController
+  def create
+    p = Page.create!(params.require(:data).require(:attributes)
```

Comment:
The API docs warn against require( ).require( ) and suggest an alternative

http://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-require

...I don't understand exactly what is being said there, but I don't find any examples of require( ).require( ) in our code either.